packages/adblocker-benchmarks/run.js

```diff
@@ -260,7 +260,7 @@ async function main() {
     }
 
     start = process.hrtime();
-    const match = engine.match({ type: WEBREQUEST_OPTIONS[cpt], frameUrl, url });
+    const match = engine.match({ type: WEBREQUEST_OPTIONS[cpt] || 'other', frameUrl, url });
```

Comment:
What is the original value of ‘cpt’ that is not in the map? Did puppeteer implement new types?

Reply:
I got `"ping"`.

Reply:
Ok, I'm thinking that maybe we should instead update `WEBREQUEST_OPTIONS` with all current puppeteer types so that this change impacts all blockers in the same way.

Reply:
This seems to be the documentation: https://github.com/puppeteer/puppeteer/blob/v10.2.0/docs/api.md It doesn't list `ping`.

Reply:
This is not making sense and I have terminated the server that generated the data. Maybe for now it's best to remove these requests.

Here's an example:

```
{
  "frameUrl": "https://www.nytimes.com/wirecutter/reviews/best-cat-litter/",
  "url": "https://a.et.nytimes.com/track",
  "cpt": "ping"
}
```

Reply:
Since `ping` is a [valid `webRequest` type](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest/ResourceType), I've added a mapping for it now.

I'd have to learn more about puppeteer to understand why it sends this type when the documentation does not list it.

Reply:
It's a bit weird because `ping` is not listed in the source code of puppeteer, *but* if you look into the definition of the devtools protocol used by puppeteer, then it's there: https://github.com/ChromeDevTools/devtools-protocol/blob/master/types/protocol.d.ts#L8908

So I guess they might just forward anything from devtools but not all of them are documented.

Reply:
There are a few here like `SignedExchange`, `CSPViolationReport`, etc. that should be mapped too. I'm not sure about how to map these though so I'll have to dig further into this.

Reply:
Sounds good! Thanks for looking into it @mjethani. In case of doubt I think it would be fine to map them to `other`, as long as we have the same behavior for all blockers.

Reply:
The only new type we've run into is `ping`, and that's also a known type as far as browser extensions are concerned. Should we really bother with the others, since we haven't run into them in more than 200,000 requests on the top sites? Because maybe puppeteer filters/maps them anyway.

Reply:
Sounds good. Let's keep it like that!